agent/kv_agent_rollout_impl.go

```diff
@@ -0,0 +1,214 @@
+package agent
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"errors"
+	"log/slog"
+	"math/big"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/kvtools/valkeyrie/store"
+	protoconfservice "github.com/protoconf/protoconf/agent/api/proto/v1"
+	protoconf_agent_config "github.com/protoconf/protoconf/agent/config/v1"
+	protoconfvalue "github.com/protoconf/protoconf/datatypes/proto/v1"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+type ProtoconfKVAgentRollout struct {
+	store       store.Store
+	config      *protoconf_agent_config.AgentConfig
+	Logger      *slog.Logger
+	sendMutex   *sync.Mutex
+	channelName string
+	agentId     string
+
+	protoconfservice.ProtoconfServiceServer
+}
+
+func NewProtoconfKVAgentRollout(store store.Store, config *protoconf_agent_config.AgentConfig) (*ProtoconfKVAgentRollout, error) {
+	_, err := store.Exists(context.Background(), "/", nil)
+	if err != nil {
+		return nil, errors.Join(errors.New("store is not available"), err)
+	}
+	return newProtoconfKVAgentRollout(store, config), nil
+}
+
+func newProtoconfKVAgentRollout(store store.Store, config *protoconf_agent_config.AgentConfig) *ProtoconfKVAgentRollout {
+	logger := slog.Default()
+	if config.ChannelName != "" {
```

Comment:
Is the ChannelName mandatory?

Reply:
No, it will be detected from the runtime environment (ENV/flags/Annotations) but can also be empty